### ISSUE TITLE
[Bug] Radio displays it has a messages but nothing plays.

### ISSUE THREAD

**Issue by ProjectPatatoe** [SubnauticaNitrox/Nitrox#1368]
### Version
1.4.0.0

### Bug
Radio shows message to be played, but i hear nothing

### Expected
Some message to be played

### Steps to reproduce
Enter game or wait a long time.

### Other details:
No one in our group (4 players) ever received the sunbeam radio events

**Comment by TeamDoodz** [comment]
When this happened to me, there was an IndexOutOfBounds exception I believe.

**Comment by Coding-Hen** [comment]
Do you have a playerlog for those IndexOutOfBounds exceptions? @TeamDoodz

**Comment by TeamDoodz** [comment]
[20201227.log](https://github.com/SubnauticaNitrox/Nitrox/files/5810830/20201227.log)

Here it is. Sorry for the late reply and the immature usernames...

**Comment by diabolical17** [comment]
Here, found the cause:

```log
17:01:55.212 [Server][ERR] Exception while processing packet: NitroxModel.Packets.RadioPlayPendingMessage System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.RemoveAt(Int32 index)
   at NitroxModel.DataStructures.ThreadSafeCollection`1.RemoveAt(Int32 index) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxModel\DataStructures\ThreadSafeCollection.cs:line 188
   at NitroxServer.RadioPlayPendingMessageProcessor.Process(RadioPlayPendingMessage packet, Player player) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxServer\Communication\Packets\Processors\RadioPlayPendingMessageProcessor.cs:line 21
   at NitroxServer.Communication.Packets.Processors.Abstract.AuthenticatedPacketProcessor`1.ProcessPacket(Packet packet, IProcessorContext player) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxServer\Communication\Packets\Processors\Abstract\AuthenticatedPacketProcessor.cs:line 10
   at NitroxServer.Communication.Packets.PacketHandler.ProcessAuthenticated(Packet packet, Player player) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxServer\Communication\Packets\PacketHandler.cs:line 46
   at NitroxServer.Communication.Packets.PacketHandler.Process(Packet packet, NitroxConnection connection) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxServer\Communication\Packets\PacketHandler.cs:line 34
   at NitroxServer.Communication.NetworkingLayer.NitroxServer.ProcessIncomingData(NitroxConnection connection, Packet packet) in C:\Users\Sunrunner\Desktop\Code\Nitrox\NitroxServer\Communication\NetworkingLayer\NitroxServer.cs:line 63
   ```

The client is trying to play a radio message that isn't in the collection.
Example: Its being told to play message 10 when there are only 5 messages.

I can get this fixed pretty quick.

**Comment by diabolical17** [comment]
Could someone assign me?

**Comment by tornac1234** [comment]
Is this still up to date ?

**Comment by MPcetacea** [comment]
We're still having this issue, is it supposed to be fixed? At first it was some messages in between but now it seems to be every message.

**Comment by tornac1234** [comment]
If you could send your logs it would help, and also send the GameData file from your save folder because imo this file may be kind of corrupted concerning story events. If you have this quick "patch" on your server, it will only make the error more readable for us.

**Comment by MPcetacea** [comment]
[game-20210815.log](https://github.com/SubnauticaNitrox/Nitrox/files/6994001/game-20210815.log)
Is this the right log? I'm also not sure where to find the GameData file. I have a BaseData, EntityData, PlayerData and a WorldData file in the world folder but I can't even find a save folder.

**Comment by MPcetacea** [comment]
[launcher-20210815.log](https://github.com/SubnauticaNitrox/Nitrox/files/6994132/launcher-20210815.log)
[server-20210815.log](https://github.com/SubnauticaNitrox/Nitrox/files/6994134/server-20210815.log)

**Comment by tornac1234** [comment]
I'll look more into it as soon as I finish working on scan progress sync. In the meanwhile, it would be great if we could obtain different logs about this issue, even though the log patch isn't included in any release.

**Comment by MPcetacea** [comment]
Thanks for putting work in on it! I'll give the GameData file whenever if you're able to help me figure out which one it is lmao. As a side note I managed to play some radio messages today when I was alone on the multiplayer server, so a hypothesis I have is that it only stops when we're more than one, but I have yet to really test it out.

**Comment by tornac1234** [comment]
@MPcetacea I can't find any trace of errors concerning radio in the latest logs you sent. I also need to know which version of Nitrox you're using. I'm guessing a bug happened and now your savefile is kind of corrupted or whatever. Did you create a brand new savefile for the 1.5 of nitrox ?

**Comment by MPcetacea** [comment]
Thanks for looking at it! We didn't start playing til after the 1.5 version as far as i know, I'm using 1.5.0.1 and have been playing around a week. 
I tried to look in the logs myself and it seemed to just... not do anything whatsoever when we click radio messages, nothing registered.
Is it possible to create a new savefile and keep the server or does that get erased then?

**Comment by tornac1234** [comment]
When i say savefile I mean your saved world containing all the infos about your current game. If you erase them you will lose your progress, so that's not the way to go. Maybe you can send me your WorldData so that I try it myself and see if I can reproduce the bug.

**Comment by tornac1234** [comment]
Quick update: I've fixed the bug as it is, but for some reason I get some radio message multiple times, which shouldn't be the case. I'm not sure if there's really a problem but if there's one it's about the fact that all clients send their radio messages when they happen to get one. In this case, they will hear the same radio message 2 times, and without the fix, the second time will trigger the radio blinking but no message will be played.

**Comment by tornac1234** [comment]
Another quick update: the radio bug is clear now:
- Each time a client spawns in the server, he will restart from the beginning concerning story goals
- When a radio call will be received, you aren't able to read it because the game considers this story goal has already been completed.

Now the solution resides in the initialisation and synchronisation of the schedule thing.

**Comment by Jannify** [comment]
Fixed with #1583
